### PR TITLE
detect single directory as app source and lift contents to match paketo's expectations

### PR DESCRIPTION
**Pull request by andreas-kupries**
ref https://github.com/epinio/epinio/issues/2465
main PR https://github.com/epinio/epinio/pull/2712

### CODE REVIEW

File: chart/epinio/templates/stage-scripts.yaml [hunk 1]
```diff
@@ -63,6 +63,17 @@ data:
...
+         # archive contains single element ...
+         if test -d "$(ls)" 2>/dev/null ; then
+           # ... which is a directory. assume that the actual app sources are
+           # inside it, and lift them out to match paketo's expectations.
+           child="$(ls)"
+           echo Lifting \"${child}\" contents ...
+           mv -v "${child}"/* "${child}"/.??* .
+           rmdir -v "${child}"
+         fi
+       fi
```
Comment (enrichman) [inline chart/epinio/templates/stage-scripts.yaml:76]: You can use `find` to check if there is only one directory:

- `-maxdepth 1` will check only the current folder
- `-mindepth 1` will exclude the `.`
- `-printf %y` will print the file type (`d` for)


```
find . -maxdepth 1 -mindepth 1 -printf %y
```

Example:

```
mkdir mydir
```
```
ls -al     
total 0
drwxr-xr-x 1 enrico enrico   10 Nov 22 13:52 .
drwxr-xr-x 1 enrico enrico 3900 Nov 22 13:53 ..
drwxr-xr-x 1 enrico enrico    0 Nov 22 13:52 mydir
```
```
find . -maxdepth 1 -mindepth 1 -printf %y
d
```

```suggestion
       # archive contains a single directory
       if [ "$(find . -maxdepth 1 -mindepth 1 -printf %y)" = "d" ] ; then
         # assume that the actual app sources are
         # inside it, and lift them out to match paketo's expectations.
         child="$(ls)"
         echo Lifting \"${child}\" contents ...
         mv -v "${child}"/* "${child}"/.??* .
         rmdir -v "${child}"
       fi
```

**Review by enrichman — COMMENTED** [review]
Small suggestion using `find`

File: chart/epinio/templates/stage-scripts.yaml (hunk 1 above)
Comment (andreas-kupries) [reply chart/epinio/templates/stage-scripts.yaml]: Done. Commit bf615fc8b7c92dcb6c744f09c6214a817d855935

**Review by enrichman — APPROVED** [review]

File: chart/epinio/templates/stage-scripts.yaml (hunk 1 above)
Comment (andreas-kupries) [reply chart/epinio/templates/stage-scripts.yaml]: Something in my implementation of this is broken. Fails to run.

Comment (andreas-kupries) [reply chart/epinio/templates/stage-scripts.yaml]: `unpack-blob CHECK=dd` ... Expected `CHECK=d` ... 

Ah: 
```
 unpack-blob .php.ini.d/ 
 unpack-blob .php.ini.d/extensions.ini 
 unpack-blob wordpress/ 
```

A hidden directory ignored by my `ls`, and not ignored the new `find`.
The fix is to undo the change to the text, i.e. go back to `BP_PHP_WEB_DIR=wordpress`.

Comment (andreas-kupries) [reply chart/epinio/templates/stage-scripts.yaml]: Done so in https://github.com/epinio/epinio/pull/2712. No changes here.

**Pull request merged**